database/schema.sql

```diff
@@ -0,0 +1,181 @@
+
+-- Valid for squlite3 3.10+ (if not exists, without rowid).
+
+-- As sql keywords are composite and have spaces inside them,
+-- I used 2 spaces to separate names, types and constraints visually.
+--
+-- Also, sqlite is said to create indices on PKs automatically, so I will omit them.
+--
+-- As the sqlite will create additional `rowid` unique PK even if another
+--  non integer or integer desc PK is present, some of tables are created
+--  WITHOUT ROWID.
+
+-- Creating 'Courses' table.
+--
+create table if not exists Courses (
+
+    -- Sqlite3 will force ascending primary key to be non-null, replacing
+    --  null with autoincremented key (while sqlite2 won't).
+    --
+    id    INTEGER,
+    desc  TEXT     null,
+
+    primary key (id asc)
+);
+
+-- Creating 'Subjects' table.
+--
+create table if not exists Subjects (
+    id         INTEGER,
+    course_id  INTEGER  not null,
+
+    primary key (id asc),
+    foreign key (course_id) references Courses (id)
+);
+
+create index if not exists Subject_course_ids on Subjects (course_id);
+
+-- Creating 'Students' table.
+--
+create table if not exists Students (
+    -- The "primary key" doesn't make field non null in sqlite.
+    --
+    addr  BLOB  not null,
+
+    primary key (addr)
+
+) without rowid;
+
+-- Creating 'StudentCourses' table.
+--
+-- [Appears to be] many-to-many ref between Students and Cources.
+--
+create table if not exists StudentCources (
+    student_addr  BLOB     not null,
+    course_id     INTEGER  not null,
+
+    -- Composite primary.
+    --
+    primary key (student_addr, course_id),
+
+    foreign key (student_addr) references Students (addr),
+    foreign key (course_id)    references Courses  (id)
+
+) without rowid;
+
+create index if not exists StudentCources_student_addr on StudentCources (student_addr);
+create index if not exists StudentCources_course_id    on StudentCources (course_id);
+
+-- Creating 'Assignments' table.
+--
+create table if not exists Assignments (
+    hash       BLOB     not null,
+    course_id  INTEGER  not null,
+    type       INTEGER  not null,
```

Comment:
We'll probably rethink structure of `Assignment`s and `Submission`s a bit (namely, we might have contents hash instead of type tag). I'll add a new comment later.